### PR TITLE
Adding build support for the ModalAI VOXL 2 platform

### PR DESCRIPTION
**Pull request by katzfey**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary. You can then open draft pull requests to get early feedback.

## Describe problem solved by this pull request
This is the first pull request to add support for the ModalAI VOXL 2 single board computer.

## Describe your solution
This is the bare minimum of code to test the build environment for VOXL 2

## Test data / coverage
A simple "Hello World" message output shows that PX4 is running

### CODE REVIEW

File: cmake/px4_add_library.cmake [hunk 1]
```diff
@@ -44,7 +44,11 @@ function(px4_add_library target)
 	target_compile_definitions(${target} PRIVATE MODULE_NAME="${target}")
 
 	# all PX4 libraries have access to parameters and uORB
-	add_dependencies(${target} uorb_headers parameters)
```
Comment (dagar) [inline cmake/px4_add_library.cmake:47]: Let me see if we can get rid of this entirely.

File: src/lib/parameters/CMakeLists.txt [hunk 2]
```diff
@@ -82,6 +82,14 @@ endif()
 if(CONFIG_BOARD_IO)
 	set(board_with_io_arg --board-with-io)
 endif()
+
+# TODO: Why do these only need to be filtered out on posix?
+if (${PX4_PLATFORM} STREQUAL "posix")
```
Comment (dagar) [inline src/lib/parameters/CMakeLists.txt:87]: What was going on here?

**Comment by hamishwillee** [comment]
When this goes in, can we have a docs addition? Like https://docs.px4.io/main/en/flight_controller/modalai_voxl_flight.html

**Comment by katzfey** [comment]
Yes, we can do that.

File: src/lib/parameters/CMakeLists.txt (hunk 2 above)
Comment (junwoo091400) [reply src/lib/parameters/CMakeLists.txt:87]: Hmmm weird

File: src/modules/muorb/apps/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+
+px4_add_module(
+	MODULE modules__muorb__apps
+	MAIN muorb
```
Comment (junwoo091400) [inline src/modules/muorb/apps/CMakeLists.txt:36]: Just curious, why is it called muorb?

File: src/lib/parameters/CMakeLists.txt (hunk 2 above)
Comment (katzfey) [reply src/lib/parameters/CMakeLists.txt]: Alright, I fixed this one and pushed up the change. It was ```set(DISABLE_PARAMS_MODULE_SCOPING TRUE)``` causing the problem.

File: src/modules/muorb/apps/CMakeLists.txt (hunk 3 above)
Comment (katzfey) [reply src/modules/muorb/apps/CMakeLists.txt:36]: multiprocessor uorb. Not my idea, was coined by Qualcomm. We can always change it to something more meaningful but maybe better to just leave it as it.

File: cmake/px4_add_library.cmake (hunk 1 above)
Comment (katzfey) [reply cmake/px4_add_library.cmake:47]: Were you able to determine if this can be removed completely?

**Comment by katzfey** [comment]
The PR got messed up by some careless commits so I submitted a fixed one and will close this one.

**Comment by katzfey** [comment]
The PR got messed up by some careless commits so I submitted a fixed one and will close this one.

**Comment by goldbattle** [comment]
PR is here: https://github.com/PX4/PX4-Autopilot/pull/20274

**Pull request closed without merging**